lr-cli/cortx_setup/commands/node/prepare/storage.py

```diff
@@ -0,0 +1,195 @@
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from cortx_setup.commands.command import Command
+from cortx_setup.commands.storage import Commons
+from cortx_setup.config import CONFSTORE_CLUSTER_FILE
+from cortx_setup.commands.common_utils import get_pillar_data
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id
+from provisioner.salt import function_run
+
+
+class NodePrepareStorage(Command):
+    """Cortx Setup API for Preparing Storage in field"""
+    _args = {
+        'user': {
+            'type': str,
+            'default': None,
+            'optional': True
+        },
+        'password': {
+            'type': str,
+            'default': None,
+            'optional': True
+        }
+    }
+
+    def validate_user_password(
+            self, targets=None, enc_num=None, user=None, passwd=None
+    ):
+
+        self.logger.debug(
+            f"Validating the provided credentials [{user}/{passwd}]"
+        )
+        _ctrl_user = get_pillar_data(
+            f'storage/{enc_num}/controller/user'
+        )
+        _ctrl_secret_enc = get_pillar_data(
+            f'storage/{enc_num}/controller/secret'
+        )
+        try:
+            # Decrypt the secret read from the storage pillar
+            _component = "storage"
+            _result = function_run('grains.get', fun_args=['cluster_id'],
+                                    targets=targets)
+            _cluster_id = _result[f'{targets}']
+        except Exception as exc:
+            self.logger.error(
+                "Could not fetch cluster id from grains"
+            )
+            raise exc
+
+        try:
+            _cipher_key = Cipher.generate_key(_cluster_id, _component)
+            _ctrl_secret = (Cipher.decrypt(
+                                _cipher_key, _ctrl_secret_enc.encode("utf-8")
+                            )).decode("utf-8")
+        except Exception as exc:
+            self.logger.error(
+                "Could not decrypt the password stored in the configuration\n"
+            )
+            raise exc
+
+        _user_check = True
+        _passwd_check = True
+
+        if user != _ctrl_user:
+            self.logger.warning(
+                f"Username provided [{user} does not match"
+                " with the user name in configuration]"
+            )
+            _user_check = False
+        else:
+            self.logger.debug(
+                f"Username provide [{user} matches with"
+                " the user name in configuration"
+            )
+
+        if passwd != _ctrl_secret:
+            self.logger.warning(
+                f"Password provided [{passwd} does not match"
```

Comment:
All warning and debug msgs have array in between `[ ]` . can be removed.

Reply:
done